### DISCUSSION TITLE
useMutation promise not resolving when offline. How do I use promises while offline?

### DISCUSSION THREAD

**Discussion by bensoutendijk** [urql-graphql/urql#2015] [Q&A]
In my project, I am trying to implement offline support. However, I must use Promises in order to avoid race conditions while online. For example, I have a form which creates a new object in the database with Urql's `useMutation`

```
Urql.useMutation(CreateScenarioDocument).finally(
  () => router.push({ name: "Scenario", params: { scenario: newScenario.id })
);
```

However, this promise seems to never be resolved while offline, so the page does not navigate after the request. How can I use the promises generated with useMutation and useQuery while offline?

**Comment by dohomi** [comment]
@bensoutendijk did you find a solution? I am wondering how to support offline. My mutation fails with `ERR_INTERNET_DISCONNECTED` even though I enabled offlineExchange
